src/app/dashboard-widgets/recent-workspaces-widget/recent-workspaces-widget.component.ts

```diff
@@ -137,9 +136,11 @@ export class RecentWorkspacesWidgetComponent implements OnDestroy, OnInit {
         return Observable.of([]);
       }
       return Observable.forkJoin(val.map((space: ExtSpace) => {
+        this.loading = true;
         return this.fetchCodebases(space.id)
           .map(codebases => {
             space.codebases = codebases;
+            this.loading = false;
```

Comment:
The loading state only turns false if the Observable can be successfully subscribed to, what happens if the error block is triggered? Or if for whatever reason the `workItemService.getWorkspaces()` [0] throws an error? It looks like the widget will continue to display a loading state? Could there be tests that verify the behaviour of a successful/unsuccessful return from the service? Does this same behaviour apply to the `work-item-widget.component.ts` changes as well?

[0] https://github.com/dlabrecq/fabric8-ui/blob/2a8630f1f60537f9cf97bbf0073aca19a47d81d1/src/app/dashboard-widgets/recent-workspaces-widget/recent-workspaces-widget.component.ts#L114

Reply:
Changed the code to set the loading state after the catch.

This PR is limited to the space subscriptions used with the home dashboard. The same behavior does not need to be applied to other widgets used on the space dashboard because that code runs within the context of a space and the loading state is set properly.